### ISSUE TITLE
[FR] Replace `reject_unknown_reverse_client_hostname` with Rspamd `HFILTER_HOSTNAME_UNKNOWN` and make it configurable

### ISSUE THREAD

**Issue by georglauterbach** [docker-mailserver/docker-mailserver#3228]
### Context

See https://github.com/docker-mailserver/docker-mailserver/issues/3224#issuecomment-1500111915.

### Relations

We are using `reject_unknown_reverse_client_hostname` in Postfix's config as a non-configurable default. This is actually mostly fine, but we had requests wanting to make this configurable.

### Description

Now, Rspamd supports something very similar with `HFILTER_HOSTNAME_UNKNOWN`. If the reverse client hostname cannot be found, a (configurable) number is added to the score. We should make this whole setting configurable too.

### Alternatives

N/A

### Applicable Users

Everyone who is using Rspamd OR those that want to configure this setting. I think adding this to the spam score is nice because the score calculation is an overall sum 
where multiple factors are accounted for, so whether an email is spam or not measured by the score should be faily accurate.

### Are you going to implement it?

Yes, because I know the probability of someone else doing it is low and I can learn from it.

### What are you going to contribute?

Everything.

### Already Done

Nothing.

**Comment by georglauterbach** [comment]
This feature is blocked by #3206 (which itself is blocked by the relase of v12.0.0 (#3146)). When #3206 is merged, implementation of this feature can start.

**Comment by georglauterbach** [comment]
This issue will also resolve #2970.
